src/AggregateFunctions/AggregateFunctionSequenceNextNode.h

```diff
@@ -0,0 +1,462 @@
+    {
+        // This aggregate function sets insertion_requires_nullable_column on.
+        // Even though some values are mapped to aggregating key, it could return nulls for the below case.
+        //   aggregated events: [A -> B -> C]
+        //   events to find: [C -> D]
+        //   [C -> D] is not matched to 'A -> B -> C' so that it returns null.
+        return std::make_shared<AggregateFunctionNullVariadic<false, false, true, true>>(nested_function, arguments, params);
+    }
+
+    void insert(Data & a, const Node * v, Arena * arena) const
+    {
+        ++a.total_values;
+        a.value.push_back(v->clone(arena), arena);
+    }
+
+    void create(AggregateDataPtr place) const override
+    {
+        [[maybe_unused]] auto a = new (place) Data;
+    }
+
+    void add(AggregateDataPtr place, const IColumn ** columns, size_t row_num, Arena * arena) const override
+    {
+        Node * node = Node::allocate(*columns[1], row_num, arena);
+
+        const auto timestamp = assert_cast<const ColumnVector<T> *>(columns[0])->getData()[row_num];
+
+        /// The events_bitset variable stores matched events in the form of bitset.
+        /// It uses UInt32 instead of std::bitset because bitsets of UInt32 are easy to compare. (< operator on bitsets)
+        /// Each Nth-bit indicates that the Nth-event are matched.
+        /// For example, event1 and event3 is matched then the values of events_bitset is 0x00000005.
+        ///   0x00000000
+        /// +          1 (bit of event1)
+        /// +          4 (bit of event3)
+        UInt32 events_bitset = 0;
+        for (UInt8 i = 0; i < events_size; ++i)
+            if (assert_cast<const ColumnVector<UInt8> *>(columns[2 + i])->getData()[row_num])
+                events_bitset += (1 << i);
+        if (events_bitset == 0) events_bitset = 0x80000000; // Any events are not matched.
+
+        node->event_time = timestamp;
+        node->events_bitset = events_bitset;
+
+        data(place).value.push_back(node, arena);
+    }
+
+    void merge(AggregateDataPtr place, ConstAggregateDataPtr rhs, Arena * arena) const override
+    {
+        if (data(rhs).value.empty())
+            return;
+
+        if (data(place).value.size() >= max_elems)
+            return;
+
+        auto & a = data(place).value;
+        auto & b = data(rhs).value;
+        const auto a_size = a.size();
+
+        const UInt64 new_elems = std::min(data(rhs).value.size(), static_cast<size_t>(max_elems) - data(place).value.size());
+        for (UInt64 i = 0; i < new_elems; ++i)
+            a.push_back(b[i]->clone(arena), arena);
+
+        /// either sort whole container or do so partially merging ranges afterwards
+        using Comparator = typename SequenceNextNodeGeneralData<T, Node, Descending>::Comparator;
+
+        if (!data(place).sorted && !data(rhs).sorted)
+            std::stable_sort(std::begin(a), std::end(a), Comparator{});
```

Comment:
The method `sort` of Data is a general sorting method. It doesn't considering whether sort is necessary or not so that I don't use the method `sort` in `merge`.
My considering is two cases.
- At the line 224~225, it perform sorting whole list when two lists are not sorted.
- The below lines 226~, it perform partial sorting for each lists, and perform `inplace_merge` to speed up.
I don't know better sorting algorithm. Please tell me some more details.